addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/start-ec2-environment/start-ec2-environment-sc.js

```diff
@@ -0,0 +1,186 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const _ = require('lodash');
+const StepBase = require('@aws-ee/base-workflow-core/lib/workflow/helpers/step-base');
+
+class StartEc2EnvironmentSc extends StepBase {
+  async start() {
+    const environmentId = await this.payload.string('environmentId');
+    this.state.setKey('STATE_ENVIRONMENT_ID', environmentId);
+
+    const requestContext = await this.payload.object('requestContext');
+    this.state.setKey('STATE_REQUEST_CONTEXT', requestContext);
+
+    const instanceId = await this.payload.string('instanceIdentifier');
+
+    const ec2 = await this.getEc2Service();
+
+    const params = {
+      InstanceIds: [instanceId],
+    };
+
+    let instanceStatusInfo;
+    try {
+      const paramsForDescribe = { ...params, IncludeAllInstances: true };
+      instanceStatusInfo = await ec2.describeInstanceStatus(paramsForDescribe).promise();
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('describe ec2 instance error: ', error);
+      throw error;
+    }
+
+    const status = _.get(instanceStatusInfo, 'InstanceStatuses[0].InstanceState.Name').toUpperCase();
+
+    if (status !== 'STOPPED') {
+      throw new Error(`EC2 instance [${instanceId}] is not stopped`);
+    }
+
+    try {
+      await ec2.startInstances(params).promise();
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('start ec2 instance error: ', error);
+      throw error;
+    }
+
+    await this.updateEnvironment({ status: 'STARTING' });
+    this.state.setKey('STATE_INSTANCE_ID', instanceId);
+
+    return this.wait(5)
+      .maxAttempts(120)
```

Comment:
That's a good question! I'll look into how maxAttempts and wait works.

Reply:
Most of the time, the AWS SDK will handle the backoff logic for you.  For example, certain AWS APIs will return errors that are marked as 'retryable'.  When the AWS SDK sees that in the exception, it will retry again with exponential backoff logic.  If you still want to introduce some randomness into the wait.  You can take a look at  addons/addon-base-raas/packages/base-raas-workflow-steps/lib/steps/delete-environment/delete-environment.js (line 76), using the fuzz utility function.

`return this.wait(fuzz(80))
      .maxAttempts(120)`